ci2/ci/github.py

```diff
@@ -105,6 +109,17 @@ def to_dict(self):
         return {'repo': self.repo.to_dict(), 'name': self.name}
 
 
+class MostRecentBuild:
+    def __init__(self, source_sha, target_sha, build_state, batch):
+        self.source_sha = source_sha
+        self.target_sha = target_sha
+        self.build_state = build_state
```

Comment:
Where do you update the build_state of a most recent batch that isn't complete?  I think this needs to happen in PR._heal where we update the build_state based on batch.